include/zephyr/sensing/sensing_sensor.h

```diff
@@ -0,0 +1,594 @@
+ *
+ * @param samples The sample number to set for fifo.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+typedef int (*sensing_sensor_set_fifo_t)(
+		const struct device *dev,
+		uint32_t samples);
+
+/**
+ * @brief Get current sensor's hardware fifo size
+ *
+ * Some sensors especially for physical sensors, support fifo, this API can
+ * get the current fifo size.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param samples The data buffer to receive the fifo sample number.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+typedef int (*sensing_sensor_get_fifo_t)(
+		const struct device *dev,
+		uint32_t *samples);
+
+/**
+ * @brief Set current sensor data offset callback
+ *
+ * Some sensors especially for physical sensors, such as accelerometer senors,
+ * as data drift, need configure offset calibration.
+ *
+ * Since each sensor type may have multiple data fields for it's value, this
+ * API support set separated offset for each data field, or global offset for
+ * all data fields.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The offset value to be set.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+typedef int (*sensing_sensor_set_offset_t)(
+		const struct device *dev,
+		int index, int32_t value);
+
+/**
+ * @brief Get current sensor data offset callback
+ *
+ * Some sensors especially for physical sensors, such as accelerometer senors,
+ * as data drift, need configure offset calibration.
+ *
+ * Since each sensor type may have multiple data fields for it's value, this
+ * API support get separated offset for each data field, or global offset for
+ * all data fields.
+ *
+ * @param dev The sensor instance device structure.
+ *
+ * @param index The value fields index to be set, -1 for all fields (global).
+ *
+ * @param value The data buffer to receive the offset value.
+ *
+ * @return 0 on success or negative error value on failure, not support etc.
+ */
+typedef int (*sensing_sensor_get_offset_t)(
+		const struct device *dev,
+		int index, int32_t *value);
+/**
+ * @struct sensing_sensor_api
+ * @brief Sensor callback api
+ *
+ * A sensor must register this callback API during sensor registration.
+ */
+struct sensing_sensor_api {
+	sensing_sensor_init_t init;
+	sensing_sensor_reset_t reset;
+	sensing_sensor_deinit_t deinit;
+	sensing_sensor_set_interval_t set_interval;
+	sensing_sensor_get_interval_t get_interval;
+	sensing_sensor_set_range_t set_range;
+	sensing_sensor_get_range_t get_range;
+	sensing_sensor_set_offset_t set_offset;
+	sensing_sensor_get_offset_t get_offset;
+	sensing_sensor_get_fifo_t get_fifo;
+	sensing_sensor_set_fifo_t set_fifo;
+	sensing_sensor_set_sensitivity_t set_sensitivity;
+	sensing_sensor_get_sensitivity_t get_sensitivity;
```

Comment:
These seem to be duplicates of the sensor API attributes. Can we leverage those instead of creating new ways?

Reply:
the `sensor API attributes` do you mean the `enum sensor_attribute` in existing `sensor device API`? (include/zephyr/drivers/sensor.h) 

Some attributes definitions in `enum sensor_attribute` like `SENSOR_ATTR_SAMPLING_FREQUENCY`, `SENSOR_ATTR_OFFSET` may similar, but `sensing subsystem` do have new. 

The most important thing is we need make clear layers, only the physical sensor implementation should talk with underlying `sensor device API`, within `sensing subsystem` we should not use definitions from `sensor device API`, otherwise it will bring many confuses, future more if in some day the `sensor device API` has updated to V2 and not used `enum sensor_attribute`, then we only need change physical sensors.    

![MicrosoftTeams-image](https://github.com/zephyrproject-rtos/zephyr/assets/45843554/bbd54735-4326-4976-ac2f-4f11246170f8)

Reply:
See comment above. This isn't needed.

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/55389/files#r1205227812